clippy_lints/src/pub_use.rs

```diff
@@ -0,0 +1,55 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use rustc_ast::ast::{Item, ItemKind, VisibilityKind};
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Restricts the usage of `pub use ...`
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This may result in unexported items
```

Comment:
Could you add more explanation? It doesn't seem clear to me that it would be better to do it like in the example.

Reply:
This is taken from the issue that the lint references, the point is to avoid manually exporting specific items from a crate but to export the crate. 

Do you have a description in mind? I can try to expand on this one

Reply:
Updated

Reply:
Updated